pandas-stubs/io/formats/style.pyi

```diff
@@ -1,97 +1,267 @@
+        bold_headers: bool = ...,
+        caption: str | None = ...,
+        max_rows: int | None = ...,
+        max_columns: int | None = ...,
+        encoding: str | None = ...,
+        doctype_html: bool = ...,
+        exclude_styles: bool = ...,
+        **kwargs,
+    ): ...
+    def set_td_classes(self, classes: DataFrame) -> Styler: ...
+    def __copy__(self) -> Styler: ...
+    def __deepcopy__(self, memo) -> Styler: ...
     def clear(self) -> None: ...
-    def apply(self, func, axis: int = ..., subset=..., **kwargs): ...
-    def applymap(self, func, subset=..., **kwargs): ...
-    def where(self, cond, value, other=..., subset=..., **kwargs): ...
-    def set_precision(self, precision): ...
-    def set_table_attributes(self, attributes): ...
-    def export(self): ...
-    def use(self, styles): ...
-    def set_uuid(self, uuid): ...
-    def set_caption(self, caption): ...
-    def set_table_styles(self, table_styles): ...
-    def set_na_rep(self, na_rep: str) -> Styler: ...
-    def hide_index(self): ...
-    def hide_columns(self, subset): ...
-    def highlight_null(self, null_color: str = ...): ...
+    def apply(
+        self,
+        func: Callable,
+        axis: Axis | None = ...,
+        subset: Subset | None = ...,
+        **kwargs,
+    ) -> Styler: ...
+    def apply_index(
+        self,
+        func: Callable,
+        axis: int | str = ...,
+        level: Level | list[Level] | None = ...,
+        **kwargs,
+    ) -> Styler: ...
+    def applymap_index(
+        self,
+        func: Callable,
+        axis: int | str = ...,
+        level: Level | list[Level] | None = ...,
+        **kwargs,
+    ) -> Styler: ...
+    def applymap(
+        self, func: Callable, subset: Subset | None = ..., **kwargs
+    ) -> Styler: ...
+    # def where(self, cond: Callable, value: str, other: str | None = ..., subset: Subset | None = ..., **kwargs) -> Styler: ...
+    # def set_precision(self, precision: int) -> StylerRenderer: ...
+    def set_table_attributes(self, attributes: str) -> Styler: ...
+    def export(self) -> dict[str, Any]: ...
+    def use(self, styles: dict[str, Any]) -> Styler: ...
+    uuid: Any  # Incomplete
+    def set_uuid(self, uuid: str) -> Styler: ...
+    caption: Any  # Incomplete
+    # def set_caption(self, caption: str | tuple) -> Styler: ...
+    def set_sticky(
+        self,
+        axis: Axis = ...,
+        pixel_size: int | None = ...,
+        levels: Level | list[Level] | None = ...,
+    ) -> Styler: ...
+    def set_table_styles(
+        self,
+        table_styles: dict[Any, CSSStyles] | CSSStyles | None = ...,
+        axis: int = ...,
+        overwrite: bool = ...,
+        css_class_names: dict[str, str] | None = ...,
+    ) -> Styler: ...
+    # def set_na_rep(self, na_rep: str) -> StylerRenderer: ...
+    # def hide_index(self, subset: Subset | None = ..., level: Level | list[Level] | None = ..., names: bool = ...,) -> Styler: ...
+    def hide_columns(
```

Comment:
this is deprecated??  docs look odd, so unclear, because it says deprecated, then documents new arguments in 1.4 !

Reply:
I think `hide_index` and `hide_columns` were both updated as part of the 1.4 release cycle, and then later within the same cycle were deprecated in favour of `hide` which combined the two (and the added args). Agree looks a bit odd.